src/Microsoft.DotNet.ProjectModel/ProjectModelPlatformExtensions.cs

```diff
@@ -28,7 +29,8 @@ private static void CollectDependencies(IDictionary<string, LibraryExport> expor
             foreach (var dependency in dependencies)
             {
                 var export = exports[dependency.Name];
-                if (export.Library.Identity.Version.Equals(dependency.VersionRange.MinVersion))
+                if (export.Library.Identity.Version.Equals(dependency.VersionRange.MinVersion) 
+                    && !exclusionList.Contains(dependency.Name))
```

Comment:
@eerhardt The contains check enforces a single visit to a node, it's not intended to guard the hashset add. 

The main perf bug here was that nodes were being walked *many* times because they were a common dependency.